docs/interactions.rst

```diff
@@ -0,0 +1,274 @@
+
+.. note::
+    Although purely functional bots are possible, its recommended to inherit from the `Client` class as seen in the example below.
+
+.. code-block:: python
+
+  from pincer import Client
+  from pincer.commands import command
+
+  class Bot(Client):
+    @command
+    async def some_command(self):
+      ...
+
+  bot = Bot("TOKEN")
+  bot.run()
+
+This registers a command called "some_command". Its pretty useless right now so lets take a closer look at what else you can do.
+
+.. note::
+    If you aren't seeing your command register, its likely because it takes up to one hour for a global slash command.
+    Specify the guild in the command decorator to have application commands register instantly.
+
+To register your command a guild do:
+
+.. code-block:: python
+
+    @command(guild=SOME_NUMBER)
+    async def some_command(self):
+        ...
+
+Sending messages to the user is extremely easy. What you want to return is inferred by the object's return type. A :class:`str` can be returned to send a text message.
+Heres a simple ping command.
+
+.. code-block:: python
+
+    @command
+    async def ping(self):
+        return "pong"
+
+
+If you need access to more information, you can pass in the :data:`ctx` object.
+
+.. note::
+    `ctx` and `self` should be those exact names or the correct value will not be passed in.
+
+.. code-block:: python
+
+    from pincer.objects import MessageContext
+    ...
+
+    @command
+    async def hello(self, ctx: MessageContext):
+        # Returns the name of the user that initiated the interaction
+        return f"hello {ctx.author}"
+
+Application Command Types
+-------------------------
+Pincer provides an API for all three interaction command types. The only thing that varies is the function signiture.
+
+.. code-block:: python
+
+    from pincer.commands import command, user_command, message_command
+    from pincer.objects import MessageContext, UserMessage, User
+    ...
+
+    @command
+    # Can have any amount of inputs
+    async def ping(self, ctx: MessageContext, arg1: str, arg2: str):
+        return "pong"
+
+    # Must have a parameter for users. User can be a GuildMember. All User
+    # methods are available to GuildMember because GuildMember inherits from
+    # User.
+    async def user_ping(self, ctx: MessageContext, user: User):
+        return "pong"
+
+    # Must have a parameter for messages.
+    async def message_ping(self, ctx: MessageContext, message: UserMessage):
+        return "pong"
+
+Interaction Timeout
+-------------------
+Interactions time out after 3 seconds. To extend the timeout to 15 minutes you can run :meth:`ack` from
+:class:`~pincer.objects.MessageContext`. :class:`~pincer.objects.app.interaction_flags.InteractionFlags` is available in this method.
+
+Arguments
+---------
+
+Arguments are more variables in the command. Notice how `word` is typehinted as string.
```

Comment:
No clue what the first sentence means really :sweat_smile:

Reply:
Changed to "Every parameter besides ``ctx`` and ``self`` is inferred to be a slash command argument."